### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by caggle**
First iteration of tests. Currently testing validity of given target.

### CODE REVIEW

**Comment by claudijd** [comment]
This PR includes changes from #1, which will need to be merged before I can review this.

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+TENABLEIO_ACCESS_KEY=
```
Comment (claudijd) [inline .env]: I'm not sure why these changes are still showing in the repo, can you try rebasing this branch?

**Comment by claudijd** [comment]
This PR says unit-tests, though I don't see any in this branch, maybe a mix up?

**Comment by caggle** [comment]
@claudijd I think I managed to do it.

Note, there are not a lot of tests at the moment, basically just test for target and target type validation.
If you have more ideas for unit tests, let me know or feel free to submit a PR.

File: tests/test_target.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from lib import target
+
+
+class TestTarget(object):
+    def test_URLtarget(self):
+        fqdnurl = "https://www.mozilla.org"
```
Comment (claudijd) [inline tests/test_target.py]: Should we be validating URLs or hostnames?   I feel like given the context for this job, we'd want hostname and not URL, and let the scan make a determination on what schemes are available

Comment (claudijd) [reply tests/test_target.py]: IMO, a URL submission should be invalid.

**Comment by claudijd** [comment]
It would also be nice to see a .travis.yml file in this repo, so we know these tests are run on each PR/commit.  Here's an example here for Python (https://github.com/claudijd/ssh_observatory/blob/master/.travis.yml)

**Comment by caggle** [comment]
> I'd like to get clarity on URL vs. hostname as a target. I feel if we accept both, we're being too loose and it adds unnecessary complexity and confuses expectations if you put in http://foo.com and get stuff that is beyond the scope of http.

I think this makes a lot of sense. It does require a logic to be implemented though, which is currently missing. Logic in the form of, FQDN -> port scan -> if 80/tcp or 443/tcp open -> Run observatory/dirb etc.
Currently the only logic is to check if SSH service is open after a port scan.

I suggest we move this into an issue of its own, rather than tracking here.

**Comment by caggle** [comment]
> I suggest we move this into an issue of its own, rather than tracking here.

Created an issue for this here: https://github.com/caggle/vautomator-standalone/issues/16.
